File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int maxCount = 1,currMaxCount = 1;
        int majElement=nums[0];
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]==nums[i-1]) 
            {
                currMaxCount++;
                if(currMaxCount>maxCount) {majElement = nums[i];maxCount=currMaxCount;}
            }
            else currMaxCount = 1;
        }
        return majElement;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ct=0,el=0,n=nums.length;
        for(int i=0;i<n;i++)
        {
            if(ct==0)
            {
                el=nums[i];
                ct++;
            }
            else if(el == nums[i]) ct++;
            else if(el!=nums[i]) ct--;
        }
        int actualCt = 0;
        for(int i=0;i<n;i++)
        {
            if(nums[i]== el ) actualCt++;
        }
        if(actualCt>n/2) return el;
        return -1; 
    }
}